### DISCUSSION TITLE
How to access end points secured by VPN

### DISCUSSION THREAD

**Discussion by hannansamad6** [cypress-io/cypress#28125] [Questions and Help]
I want to access end points which are only accessible via vpn. When I try to request them running tests in sauce/lambda test I get a 502. Running the test on cypress runner on local machine passes the test as I am connected to vpn.

**Comment by portyu9** [comment]
Hi @hannansamad6 

This is a networking issue rather than a Cypress configuration issue.

Your local run works because the machine running Cypress has a route to those endpoints through your VPN. A browser/test runner running in Sauce Labs or LambdaTest is on **their cloud network**, so it does not inherit your machine's VPN connection. That explains the `502`.

You need one of these approaches:

- **Sauce Labs:** run a Sauce Connect tunnel from a machine that can reach the VPN/private endpoints.
- **LambdaTest:** run their local/private-network tunnel from a machine that can reach those endpoints.
- Alternatively, run Cypress on a self-hosted CI runner that is itself connected to the VPN/private network.

Whichever tunnel host you use must also be able to resolve any private DNS names and connect to the endpoint directly.

Options such as `chromeWebSecurity: false` will not solve this because the problem is network routing, not browser same-origin security.

Please mark this answer as accepted if it helped, thank you.
